sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/Utils.java

```diff
@@ -560,6 +565,15 @@ public ValueHolder(V v) {
         }
     }
 
+    public static <T> T parse(byte[] item, Class<T> itemClassType, EncryptionOptions encryptionOptions) {
+
+        try {
+            return getSimpleObjectMapper().readValue(item, itemClassType);
+        } catch (IOException e) {
+            throw new IllegalStateException("Failed to get POJO.", e);
```

Comment:
Why are these modeled as Illegalstate?

Reply:
We are currently (even prior to this PR) were throwing IllegalStateException on json parsing failure.

I agree we need to change this everywhere. Will track this.